### PR TITLE
un-export (some) Cluster methods

### PR DESCRIPTION
**Pull request by travisturner**
## Overview

This PR un-exports all methods on `Cluster` except:
```
func (c *Cluster) ReceiveEvent(e *NodeEvent) error
func (c *Cluster) SetState(state string)
func (c *Cluster) State() string
func (c *Cluster) Status() *internal.ClusterStatus
```
Those still have a few dependencies in the tests that need to be addressed.

@codysoyland let me know if this PR causes too many conflicts with the work you're doing.

## Pull request checklist

- [x] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [x] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [x] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [x] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [x] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

File: cluster.go [hunk 1]
```diff
@@ -277,35 +277,35 @@ func NewCluster() *Cluster {
...
 		Logger: NopLogger,
 	}
 }
 
-// Coordinator returns the coordinator node.
-func (c *Cluster) CoordinatorNode() *Node {
-	return c.nodeByID(c.Coordinator)
+// coordinatorNode returns the coordinator node.
+func (c *Cluster) coordinatorNode() *Node {
+	return c.unprotectedNodeByID(c.Coordinator)
```
Comment (jaffee) [inline cluster.go]: OK, I think we're missing a lot of locking in Cluster.... I know this PR is a pretty straight up renaming, so I'll ignore it for now.

File: cluster.go [hunk 2]
```diff
@@ -511,11 +511,6 @@ func (c *Cluster) ReceiveNodeState(nodeID string, state string) error {
 	return nil
 }
 
-// localNode is not being used.
-//func (c *Cluster) localNode() *Node {
```
Comment (jaffee) [inline cluster.go]: remove?

File: cluster.go (hunk 1 above)
Comment (travisturner) [reply cluster.go]: agreed. i’ll make a ticket for that

File: cluster.go (hunk 2 above)
Comment (jaffee) [reply cluster.go]: I are dum

**Comment by travisturner** [comment]
@jaffee I fixed the merge conflicts from the http inversion work and updated this PR.

**Pull request merged**